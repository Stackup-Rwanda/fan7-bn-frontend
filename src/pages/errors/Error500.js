import React from 'react';
import { Link } from 'react-router-dom';

const Error500 = () => (
  <div className="error">
    <div className="content">
      <h1>500</h1>
      <div className="desc">500 Error Page</div>
      <div className="actions">
        <Link to="/">
          <button type="button">Back to Home</button>
        </Link>
      </div>
    </div>
  </div>
);

export default Error500;
