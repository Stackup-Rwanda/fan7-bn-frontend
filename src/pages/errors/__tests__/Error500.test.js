import React from 'react';
import { mount, shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import Error500 from '../Error500';

const history = {
  push: jest.fn(),
  replace: jest.fn(),
  createHref: jest.fn(),
};

describe('Error500 Component Test', () => {
  it('should return true if component exists', () => {
    const wrapper = mount(
      <Router history={history}>
        <Error500 />
      </Router>,
    );

    expect(wrapper.find('.error').exists()).toBe(true);
    expect(wrapper.find('h1').text()).toEqual('500');
  });
});

describe('<Error500 /> shallow rendering tests', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<Error500 />);
    expect(tree).toMatchSnapshot();
  });
});
