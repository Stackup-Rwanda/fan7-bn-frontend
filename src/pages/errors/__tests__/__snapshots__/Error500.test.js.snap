// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error500 /> shallow rendering tests matches the snapshot 1`] = `
<div
  className="error"
>
  <div
    className="content"
  >
    <h1>
      500
    </h1>
    <div
      className="desc"
    >
      500 Error Page
    </div>
    <div
      className="actions"
    >
      <Link
        to="/"
      >
        <button
          type="button"
        >
          Back to Home
        </button>
      </Link>
    </div>
  </div>
</div>
`;
